addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -595,6 +595,16 @@ const wSnippetMenu = weSnippetEditor.SnippetsMenu.extend({
     _allowParentsEditors($snippet) {
         return this._super(...arguments) && !$snippet[0].classList.contains("s_popup");
     },
+    /**
+     * @override
+     */
+    _adaptSnippetContent($target) {
+        // Build the highlighted text content for the snippets.
+        for (const textEl of $target[0]?.querySelectorAll(".o_text_highlight") || []) {
+            applyTextHighlight(textEl);
+        }
+        return this._super(...arguments);
+    },
```

Comment:
I am a bit confused as to why it is not the public widget which handles this? And will it handle resize etc if you `applyTextHighlight` like this outside of the public widget?

Reply:
The public widget is individually observing each highlighted text (only the elements requiring highlight adaptation, since we don't care about a "resize" as long as it doesn't affect the highlights… see: `_observeHighlightResize()`). And once a highlight is applied the first time on a text, it will be detected and adapted by the public widget on every change (see: `"text_highlight_added"`) .

Remarks: 
- Even if we update the code in public widget to observe the `wrapwrap`, we'll still have some non-handled situations, like when a small snippet (that won't change the size of the `editable` container) is dropped.
- Starting from `18.0`, this fix will be moved to the `AddSnippetDialog` component code (probably after the `props.addSnippet()` action) :+1: .